packages/provider/src/providers/sequence-web3-provider.ts

```diff
@@ -0,0 +1,61 @@
+import { Web3Provider, AsyncSendable, TransactionRequest, TransactionResponse, JsonRpcSigner } from "ethers/providers"
+import { Networkish, BigNumberish, Interface } from "ethers/utils"
+import { toArcadeumTransactions, makeExpirable, makeAfterNonce, arcadeumTxAbiEncode } from "../utils"
+import { NonceDependency, ArcadeumContext } from "../types"
+import { abi as mainModuleAbi } from '../abi/mainModule'
+import { ethers, Signer } from "ethers"
+
+export class SequenceWeb3Provider extends Web3Provider {
```

Comment:
let's call this still ArcadeumWeb3Provider, to be consistent with naming until we switch everything one-shot -- makes it less confusing I think